### ISSUE TITLE
Postfix rejects external mails.

### ISSUE THREAD

**Issue by D3luxee** [mailcow/mailcow-dockerized#985]
Hello everyone,

I'm using the latest version of mailcow, i can send mails between the different mailboxes and domains on this mailcow instance, but all external mails get rejected with following message / behavior.

`

31.1.2018, 19:14:58 | info | disconnect from mx01.kthx.at[212.129.56.246] ehlo=2 starttls=1 mail=1 rcpt=0/1 data=0/1 rset=1 quit=1 commands=6/8
-- | -- | --
31.1.2018, 19:14:58 | info | NOQUEUE: reject: RCPT from mx01.kthx.at[212.129.56.246]: 554 5.7.1 <mx01.kthx.at[212.129.56.246]>: Client host rejected: Access denied; from=<****@kthx.at> to=<*****@neuland.tech> proto=ESMTP helo=<mx01.kthx.at>
31.1.2018, 19:14:58 | info | Anonymous TLS connection established from mx01.kthx.at[212.129.56.246]: TLSv1.2 with cipher DHE-RSA-CAMELLIA256-SHA256 (256/256 bits)
31.1.2018, 19:14:58 | info | connect from mx01.kthx.at[212.129.56.246]

`

I can still send mails without any issues, `docker-compose down && docker-compose up -d` was'nt able to fix this issue.

Currently i guess its related to the unbound or the dns blacklists, in this moment i received a mail (ad) then i tested it with my gmail and posteo account and still get rejected.. weird.

Any idea why this happens or where i could find more detailed logs?

**Comment by andryyy** [comment]
"Backup MX" is not checked in mailcow UI, right?

Do you send from a domain that is handled in mailcow?

**Comment by D3luxee** [comment]
Yes, "Backup MX" is not checked and i can send mails from domains / mailboxes handled by mailcow to other internal mailboxes and also external mailboxes.

Those internal messages can also be received but all mails from external servers are rejected.

Thanks for your fast response

**Comment by andryyy** [comment]
Are you trying to send mails from domains to your mailcow server, that you also created in mailcow?

**Comment by D3luxee** [comment]
No, basically mailcow rejects incoming mails no matter from where they come.

It rejected legit mails from big providers like gmail.com, gmx.com, web.de, t-online.de.

Outgoing mails from mailcow are no problem.

**Update:**
I've created a testing mailbox: 

`test@neuland.tech`
Can someone send a test mail to this address, so i can find out if its just common domains or even new domains / servers.

**Comment by BonGoo** [comment]
![obraz](https://user-images.githubusercontent.com/26207851/35644717-0b462010-06ca-11e8-8817-f6358b2bd098.png)
This is errors from hunter.io maybe this will help you to repair it.

**Comment by levinuss** [comment]
This is what checktls.com says: https://gist.github.com/levinus/32bf3371df08dbb3f0e35c808c1e3af5#file-checktls-log-L67
I have actually no idea what could be the problem here. My mailcow gives me the exact same "access denied" message when I try to send you a mail.

**Comment by D3luxee** [comment]
This is what happens on my end:
`

31.1.2018, 21:16:47 | info | NOQUEUE: reject: RCPT from mail.levinus.de[5.1.93.5]: 554 5.7.1 <mail.levinus.de[5.1.93.5]>: Client host rejected: Access denied; from=<admin@levinus.de> to=<test@neuland.tech> proto=ESMTP helo=<mail.levinus.de>
-- | -- | --
31.1.2018, 21:14:26 | info | NOQUEUE: reject: RCPT from www6.checktls.com[159.89.187.50]: 554 5.7.1 <www6.checktls.com[159.89.187.50]>: Client host rejected: Access denied; from=<test@checktls.com> to=<test@neuland.tech> proto=ESMTP helo=<checktls.com>
31.1.2018, 21:08:39 | info | NOQUEUE: reject: RCPT from www6.checktls.com[159.89.187.50]: 554 5.7.1 <www6.checktls.com[159.89.187.50]>: Client host rejected: Access denied; from=<test@checktls.com> to=<test@neuland.tech> proto=ESMTP helo=<checktls.com>
31.1.2018, 21:16:47 | info | NOQUEUE: reject: RCPT from mail.levinus.de[5.1.93.5]: 554 5.7.1 <mail.levinus.de[5.1.93.5]>: Client host rejected: Access denied; from=<admin@levinus.de> to=<test@neuland.tech> proto=ESMTP helo=<mail.levinus.de>
31.1.2018, 21:14:26 | info | NOQUEUE: reject: RCPT from www6.checktls.com[159.89.187.50]: 554 5.7.1 <www6.checktls.com[159.89.187.50]>: Client host rejected: Access denied; from=<test@checktls.com> to=<test@neuland.tech> proto=ESMTP helo=<checktls.com>
31.1.2018, 21:08:39 | info | NOQUEUE: reject: RCPT from www6.checktls.com[159.89.187.50]: 554 5.7.1 <www6.checktls.com[159.89.187.50]>: Client host rejected: Access denied; from=<test@checktls.com> to=<test@neuland.tech> proto=ESMTP helo=<checktls.com>

`
Diff master to local: [https://pastebin.com/fE66xSu4](https://pastebin.com/fE66xSu4)

DNS Check by mailcow: [https://image.ibb.co/iBFozR/2018_01_31_212345_588x914_scrot.png](https://image.ibb.co/iBFozR/2018_01_31_212345_588x914_scrot.png)

**Comment by andryyy** [comment]
Can you mail me access to info@servercow.de ? AnyDesk or similar is fine. Let's fix it.

**Comment by D3luxee** [comment]
I've sent you the mail, thanks in advance.

**Comment by D3luxee** [comment]
Hello,

I've debugged this issue together with some friends and we traced it back to following loglines.

```
rspamd-mailcow_1     | 2018-02-02 17:58:24 #14(normal) <rznbhc>; map; rspamd_map_read_cached: http://nginx:8081/forwardinghosts.php: read map data cached 102 bytes
rspamd-mailcow_1     | 2018-02-02 17:58:52 #12(controller) <9fydgy>; map; http_map_finish: http://172.22.1.251:8081/settings.php(172.22.1.251:8081): read map data 472 bytes
```
As you might notice rspamd is using the hostname nginx once to call the forwardinghosts.php which resolves in my case to 172.22.1.11 but then its trying to connect to the same nginx container with a hardcoded ip "172.22.1.251:8081" 

I've set the nginx container ip statically to 172.22.1.251 and afterwards everything is working again.

I've also searched for the ip 172.22.1.251 with rgrep but found nothing, so i guess its some binary format or even part of the database.

Maybe relates to this commit: https://github.com/mailcow/mailcow-dockerized/commit/83a21259f7269cc5eb1882590263f36455fd3754

Maybe you should implement a fallback in cases where rspamd fails to not reject but rather skip the spam detection.

**Update**

While debugging i've set the rspamd container to a older version which caused this behavior i described above, anyways its working again.

idk what's the real cause of this behavior but thank you guys.
